jcl/src/java.base/share/classes/java/security/AccessController.java

```diff
@@ -486,6 +487,9 @@ private static AccessControlContext getContextHelper(boolean forDoPrivilegedWith
 		}
 		if (null != acc && null != acc.domainCombiner) {
 			accTmp.domainCombiner = acc.domainCombiner;
```

Comment:
Is this assignment still required, considering dc will always be used?

Reply:
It is possible that there are multiple `AccessControlContext` object discovered during stack walking along with multiple `non-null domainCombiner`. Returning `AccessControlContext.domainCombiner` will be set to the `domainCombiner` from `AccessControlContext` in the closed frame. For other cases, I feel the `domainCombiner` will be set as before.
One optimization could be as following:
```
			if (dc == null) {
				dc = acc.domainCombiner;
			} else {
				accTmp.domainCombiner = acc.domainCombiner;
			}
``` 
This saves a unnecessary assignment. 
PR is updated.